js/bundle.go

```diff
@@ -39,6 +40,23 @@ type Bundle struct {
 	moduleResolver  *moduleResolver
 }
 
+// TODO: this is to be removed once this is not a warning and it can be moved to the registry
+func (b *Bundle) checkMetricNamesForPrometheusCompatibility() {
+	const (
+		nameRegexString = "^[a-zA-Z_][a-zA-Z0-9_]{1,128}$"
+		badNameWarning  = "Metric name should only include ASCII letters, numbers and underscores. " +
+			"This name will stop working in k6 v0.48.0 (around December 2023)."
+	)
+
+	compileNameRegex := regexp.MustCompile(nameRegexString)
```

Comment:
We should compile this regex only once. Currently it's done for every new bundle instance.

Reply:
I can move it - but in a real k6 instance this happens *once*. Which is also why I chose to move it here specifically as I know this is only called once, so the logs will happen only once and I don't need to do anykind of additional work.

I am okay with moving outside, it will slightly speed up *some* tests :shrug: . My main reason is that in 6 months we will be able to just delete this, whcih seemed slightly cleaner to me

Reply:
Ah, my bad. I had forgotten `newBundle` is indeed only called once. Nevermind, this is fine then.